d4s2_api_v2/api.py

```diff
@@ -138,3 +141,65 @@ def current_user(self, request):
         current_user = self.request.user
         serializer = UserSerializer(self.request.user)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class S3EndpointViewSet(viewsets.ReadOnlyModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = S3EndpointSerializer
+    queryset = S3Endpoint.objects.all()
+
+
+class S3UserViewSet(viewsets.ReadOnlyModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = S3UserSerializer
+
+    def get_queryset(self):
+        """
+        Allows filtering by email query param only showing normal users
+        """
+        email = self.request.query_params.get('email')
+        queryset = S3User.objects.filter(type=S3UserTypes.NORMAL)
+        if email:
+            return queryset.filter(user__email=email)
+        return queryset
+
+
+class S3BucketViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = S3BucketSerializer
+
+    def get_queryset(self):
+        """
+        Users can only see buckets they own or a user received a a delivery
+        """
+        return S3Bucket.objects.filter(
+            Q(owner__user=self.request.user) |
+            Q(deliveries__to_user__user=self.request.user)
+        )
+
+
+class S3DeliveryViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = S3DeliverySerializer
+
+    def get_queryset(self):
+        """
+        Users can only see the deliveries they sent or received.
+        """
+        return S3Delivery.objects.filter(Q(from_user__user=self.request.user) | Q(to_user__user=self.request.user))
+
+    @detail_route(methods=['POST'])
+    def send(self, request, pk=None):
+        s3_delivery = self.get_object()
+        if not s3_delivery.is_new() and not get_force_param(request):
+            raise AlreadyNotifiedException(detail='S3 Delivery already in progress')
+        # TODO create app to handle s3ownership UI
+        # accept_path = reverse('s3ownership-prompt') + "?s3_delivery_id=" + str(s3_delivery.id)
+        # accept_url = request.build_absolute_uri(accept_path)
```

Comment:
Placeholder code for when we add a django app to allow a user to handle accepting a delivery.